arch/arm/core/cortex_m/prep_c.c

```diff
@@ -23,6 +23,44 @@
 #include <kernel_internal.h>
 #include <arch/arm/cortex_m/cmsis.h>
 #include <string.h>
+#include <cortex_m/stack.h>
+
+static inline void switch_sp_to_psp(void)
+{
+	__set_CONTROL(__get_CONTROL() | CONTROL_SPSEL_Msk);
+	/*
+	 * When changing the stack pointer, software must use an ISB instruction
+	 * immediately after the MSR instruction. This ensures that instructions
+	 * after the ISB instruction execute using the new stack pointer.
+	 */
+	__ISB();
+}
+
+static inline void set_and_switch_to_psp(void)
+{
+	uint32_t process_sp = (uint32_t)&_interrupt_stack;
```

Comment:
u32_t (and in all similar places)
Also, you might be getting a warning for missing an empty line between declaration of variable (in line 41) and line 42

Reply:
my bad, let me fix all uint32_t to u32_t in v3.
No, no such warning.

Reply:
you were right, just ran checkpatch which gave me the warning.
I will fix it in v3.